src/fmu/dataio/_export_item.py

```diff
@@ -13,21 +13,45 @@
 from . import _utils
 
 VALID_SURFACE_FORMATS = {"hdf": ".hdf", "irap_binary": ".gri"}
+VALID_GRID_FORMATS = {"hdf": ".hdf", "roff": ".roff"}
 VALID_TABLE_FORMATS = {"hdf": ".hdf", "csv": ".csv"}
 VALID_POLYGONS_FORMATS = {"hdf": ".hdf", "csv": ".csv"}
 
-ALLOWED_CONTENTS = [
-    "depth",
-    "time",
-    "seismic",
-    "fluid_contact",
-    "volumetrics",
-    "undefined",
-]
+# the content must conform with the given json schema, e.g.
+# https://github.com/equinor/fmu-metadata/blob/dev/definitions/*/schema/fmu_results.json
+#
+# When value is None, a repeat field shall not be present, otherwise it may be as this:
+# content: seismics
+# seismics:
+#   attribute: mean
+#   zrange: 42.0
+#   filter_size: 4.0
+#   scaling_factor: 1.5
+
+ALLOWED_CONTENTS = {
+    "depth": None,
+    "time": None,
+    "property": {"attribute": str, "is_discrete": bool},
+    "seismic": {
+        "attribute": str,
+        "zrange": float,
+        "filter_size": float,
+        "scaling_factor": float,
+    },
+    "fluid_contact": {"contact": str},
+    "field_outline": {"contact": str},
+    "volume": None,
+    "volumetrics": None,  # or?
+    "undefined": None,
```

Comment:
What is the rationale behind allowing "undefined" as content?

Reply:
Just temporary, noted